lms/templates/instructor/instructor_dashboard_2/make_invoice_transaction_modal.html

```diff
@@ -0,0 +1,121 @@
+                            </label>
+                            <label>
+                                <input class="radio-custom" type="radio" name="amount_type"
+                                       value="refund">${_("Refund")}
+                            </label>
+                        </li>
+
+                        <li class="field" id="make_invoice-modal-field-description">
+                            <label for="comments">${_("Description")}</label>
+                            <textarea class="field" id="comments" type="text" name="comments" value=""
+                                      aria-describedby="pwd_reset_email-tip" aria-required="true"> </textarea>
+                        </li>
+
+                        <li class="field" id="make_invoice-modal-field-course_id">
+                            <input class="field hidden" id="course_id" type="text" name="course_id"
+                                   value="${section_data['course_id'] | h}"
+                                   readonly aria-required="true"/>
+                        </li>
+                    </ol>
+                </fieldset>
+
+                <div class="submit">
+                    <input name="submit_btn" type="button" id="add_invoice_button" value="${_('Add Invoice Transaction')}"/>
+                </div>
+            </form>
+        </div>
+    </div>
+</section>
+
+<script>
+    $(function () {
+        $("#add_invoice_button").click(function (e) {
+
+            var invoice_id = $.trim($("#invoice_id").val());
+            var amount = $.trim($("#amount").val());
+            if (!($.isNumeric(invoice_id))) {
+                 $('h3#error-msgs-invoice').html('Please enter invoice id.').show();
+                return false
+             }
+            if (!($.isNumeric(amount))) {
+                $('h3#error-msgs-invoice').html('Please enter valid numeric value in amount.').show();
+                return false
+            }
+            $('h3#error-msgs-invoice').hide();
+            $("#add_invoice_button").attr('disabled', true);
+            $.ajax({
+             type: "POST",
+             data: {
+               "course_id"  :$('#course_id').val(),
+               "invoice_id": invoice_id,
+               "amount": amount,
+               "amount_type": $('input[name="amount_type"]:checked').val(),
+               "comments": $("#comments").val()
+             },
+             url: "${ section_data['make_invoice_transaction_url'] }",
+             success: function (data) {
+                 location.reload(true);
+               },
+             error: function(jqXHR, textStatus, errorThrown) {
+                var data = $.parseJSON(jqXHR.responseText);
+                $('h3#error-msgs-invoice').html(data.message).show();
+                $("#add_invoice_button").removeAttr('disabled');
+             }
+           });
+
+        });
+     });
+
+</script>
```

Comment:
This script should be included as a separate JS file so it can be cached.  Take a look at `instructor_dashboard_2.html` (near the top) for an example of how to do this.  `ecommerce.js` is a good example.